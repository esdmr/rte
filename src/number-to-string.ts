import assert from './assert.js';

const numberFormat = /* @__PURE__ */ new Intl.NumberFormat('en-US', {
	maximumFractionDigits: 20,
	notation: 'standard',
	useGrouping: false,
});

export const numberToString = (number: number) => {
	assert(Number.isFinite(number), 'number must be finite');

	return numberFormat.format(number);
};
