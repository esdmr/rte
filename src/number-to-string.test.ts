import {describe, expect, it} from 'vitest';
import {numberToString} from './number-to-string.js';

describe('numberToString', () => {
	it('rounds very small numbers', () => {
		expect(numberToString(1e-21)).toBe('0');
	});

	it('writes small numbers in digits', () => {
		expect(numberToString(1e-20)).toBe('0.00000000000000000001');
	});

	it('writes large numbers in digits', () => {
		expect(numberToString(Number.MAX_VALUE)).toMatch(
			/^179769313486231570{292}$/,
		);
	});

	it('throws if number is infinite or NaN', () => {
		for (const value of [
			Number.NEGATIVE_INFINITY,
			Number.NaN,
			Number.POSITIVE_INFINITY,
		]) {
			expect(() => numberToString(value)).toThrowErrorMatchingSnapshot(
				`${value}`,
			);
		}
	});
});
