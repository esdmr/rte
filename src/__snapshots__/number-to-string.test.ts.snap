// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`numberToString > throws if number is infinite or NaN > -Infinity 1`] = `"Assertion failed: number must be finite"`;

exports[`numberToString > throws if number is infinite or NaN > Infinity 1`] = `"Assertion failed: number must be finite"`;

exports[`numberToString > throws if number is infinite or NaN > NaN 1`] = `"Assertion failed: number must be finite"`;
